src/main/java/graphql/schema/SingletonPropertyDataFetcher.java

```diff
@@ -15,7 +15,18 @@ public class SingletonPropertyDataFetcher<T> implements LightDataFetcher<T> {
 
     private static final SingletonPropertyDataFetcher<Object> SINGLETON_FETCHER = new SingletonPropertyDataFetcher<>();
 
-    private static final DataFetcherFactory<?> SINGLETON_FETCHER_FACTORY = environment -> SINGLETON_FETCHER;
+    private static final DataFetcherFactory<?> SINGLETON_FETCHER_FACTORY = new DataFetcherFactory<Object>() {
+        @SuppressWarnings("deprecation")
+        @Override
+        public DataFetcher<Object> get(DataFetcherFactoryEnvironment environment) {
+            return SINGLETON_FETCHER;
+        }
+
+        @Override
+        public DataFetcher<Object> get(GraphQLFieldDefinition fieldDefinition) {
+            return SINGLETON_FETCHER;
+        }
+    };
```

Comment:
because I didnt override the default `public DataFetcher<Object> get(GraphQLFieldDefinition fieldDefinition)` then it returned null and hence it went to the older more heavy weight object and a `DataFetcherFactoryEnvironment` needed to be allocated.

This now means that no `DataFetcherFactoryEnvironment` objects are allocated for property fetchers